added translation expansion code (#1676)
plone/plone.restapi: docs/source/endpoints/translations.md

:language: http
```

> 

Reviewer: Please do not remove the extra blank lines because they help authors identify headings in a wall of text.

:language: http
```

> 

Reviewer: Revert.

:language: http
```

## Expansion

This endpoint can be used with the {doc}`expansion` mechanism which allows getting additional information about a content item in one query, avoiding unnecessary requests.

If a simple `GET` request is done on the content item, a new entry will be shown on the `@components` entry, with the URL of the `@translations` endpoint:

```{eval-rst}
..  http:example:: curl httpie python-requests

>     :request: ../../../src/plone/restapi/tests/http-examples/translations_expand_get.req

Reviewer: Where is this file?

```

> ```{literalinclude} ../../../src/plone/restapi/tests/http-examples/translations_expand_get.resp

Reviewer: Where is this file?

:language: http
```

> 

Reviewer: Revert.

## Expansion

This service can be used with the {doc}`expansion` mechanism which allows getting additional information about a content item in one query, avoiding additional requests.

> As soon as plone.app.multilingual is installed, a content item is translatable.
> Which means it provides the ITranslatable interface.
> Therefore the content information provided by the content endpoint is expandable to translations.

Reviewer 2: ```suggestion
Translation information can be provided by the API expansion for translatable content items.
A content type is translatable if it has the plone.translatable behavior enabled. plone.app.multilingual enables this behavior for all content types at the time that it is installed.
For other content types added later it would be necessary to enable the behavior for them also.
```

Integration of @davisagli 's suggestion

## Expansion

This service can be used with the {doc}`../usage/expansion` mechanism which allows getting additional information about a content item in one query, avoiding additional requests.

Translation information can be provided by the API expansion for translatable content items.
A content type is translatable if it has the `plone.translatable` behavior enabled. 
`plone.app.multilingual` enables this behavior for all content types at the time that it is installed.
For other content types added later, it would be necessary to enable the behavior for them also.

If a simple `GET` request is done on the content item, a new entry will be shown on the `@components` entry, with the URL of the `@translations` endpoint:

```{eval-rst}
..  http:example:: curl httpie python-requests
    :request: ../../../src/plone/restapi/tests/http-examples/translations_unexpand_get.req
```

```{literalinclude} ../../../src/plone/restapi/tests/http-examples/translations_unexpand_get.resp
:language: http
```

> 

Reviewer 2: ```suggestion
In order to expand and embed the translations component, use the GET parameter expand with the value 'translations'.
```

## Expansion

This service can be used with the {doc}`../usage/expansion` mechanism which allows getting additional information about a content item in one query, avoiding additional requests.

Translation information can be provided by the API expansion for translatable content items.
A content type is translatable if it has the `plone.translatable` behavior enabled. 
`plone.app.multilingual` enables this behavior for all content types at the time that it is installed.
For other content types added later, it would be necessary to enable the behavior for them also.

If a simple `GET` request is done on the content item, a new entry will be shown on the `@components` entry, with the URL of the `@translations` endpoint:

```{eval-rst}
..  http:example:: curl httpie python-requests
    :request: ../../../src/plone/restapi/tests/http-examples/translations_unexpanded_get.req
```

```{literalinclude} ../../../src/plone/restapi/tests/http-examples/translations_unexpanded_get.resp
:language: http
```

> In order to expand and embed the translations component, use the GET parameter expand with the value 'translations'.

Reviewer: ```suggestion
In order to expand and embed the translations component, use the GET parameter `expand` with the value `translations`.
```

## Expansion

This service can be used with the {doc}`expansion` mechanism which allows getting additional information about a content item in one query, avoiding additional requests.

As soon as plone.app.multilingual is installed, a content item is translatable.

> Which means it provides the ITranslatable interface.

Reviewer 3: A content type is translatable if it has the `plone.translatable` behavior enabled. plone.app.multilingual enables this behavior for all content types at the time that it is installed. But if other content types are added later then it would be necessary to enable the behavior for them also.

This service can be used with the {doc}`expansion` mechanism which allows getting additional information about a content item in one query, avoiding additional requests.

Translation information can be provided by the API expansion for translatable content items.

> A content type is translatable if it has the plone.translatable behavior enabled. plone.app.multilingual enables this behavior for all content types at the time that it is installed.

Reviewer: MyST markup and one sentence per line.

```suggestion
A content type is translatable if it has the `plone.translatable` behavior enabled.
`plone.app.multilingual` enables this behavior for all content types at the time that it is installed.
```

> For other content types added later it would be necessary to enable the behavior for them also.

Reviewer: ```suggestion
For other content types added later, it would be necessary to enable the behavior for them also.
```

> This endpoint can be used with the {doc}`expansion` mechanism which allows getting additional information about a content item in one query, avoiding unnecessary requests.

Reviewer 2: ```suggestion
This service can be used with the {doc}`expansion` mechanism which allows getting additional information about a content item in one query, avoiding additional requests.
```

> If a simple `GET` request is done on the content item, a new entry will be shown on the `@components` entry, with the URL of the `@translations` endpoint:

Reviewer 2: ```suggestion
As soon as plone.app.multilingual is installed, a content item is translatable.
Which means it provides the ITranslatable interface.
Therefore the content information provided by the content endpoint is expandable to translations.

If a simple `GET` request is done on a translatable content item, an additional entry is shown on the `@components` entry, with the URL of the `@translations` endpoint:
```

```{eval-rst}
..  http:example:: curl httpie python-requests

>     :request: ../../../src/plone/restapi/tests/http-examples/translations_expand_get.req

Reviewer 2: Here first a simple GET without expansion to show the 'translations' part ( with only '@id') unexpanded.
Use the  '/plone/en/test-document?expand=translations' without the '@' sign and no request body below.

Reviewer 2: Would you please first provide a simple request and response for the unexpanded case.
Remember to remove the body of the GET request.

Author: Sorry I am unable to understand much, but what I get is that I have to make 2 get requests, one without the expansion to this `plone/en/test-document?@translations` and another get request to `plone/en/test-document?expand=translations` without  `@` sign. In both the request, I don't have to provide the body of get. 
right? @ksuess

Reviewer: @Akshat2Jain HTTP examples of expansions are here:

- https://6.docs.plone.org/plone.restapi/docs/source/endpoints/navroot.html#expansion

...and docs of expansions:

- https://6.docs.plone.org/plone.restapi/docs/source/usage/expansion.html

```

```{literalinclude} ../../../src/plone/restapi/tests/http-examples/translations_expand_get.resp
:language: http

> ```

Reviewer 2: Add here:
```
A request to provide the content information plus the translations information can be down with
'/plone/en/test-document?expand=translations'
This expands the 'translations'  part of '@components'.
```

TODO Generate a response with translations in test. Then add request and response here.

This service can be used with the {doc}`expansion` mechanism which allows getting additional information about a content item in one query, avoiding additional requests.

Translation information can be provided by the API expansion for translatable content items.
A content type is translatable if it has the `plone.translatable` behavior enabled. 

> plone.app.multilingual enables this behavior for all content types at the time that it is installed.

Reviewer: This should be an inline literal.

```suggestion
`plone.app.multilingual` enables this behavior for all content types at the time that it is installed.
```
